src/Microsoft.Win32.Registry/tests/RegistryKey/RegistryKey_GetValue_CorruptData.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.Win32.RegistryTests
+{
+    public class RegistryKey_GetValue_CorruptData : RegistryTestsBase
+    {
+        [Fact]
+        public void ReadRegMultiSzLackingFinalNullTerminatorCorrectly()
+        {
+            // We have to intentionally write a REG_MULTI_SZ value
+            // lacking a final null terminator, like "str1\0str2\0str3\0"
+            // (there should be 2 \0s at the end) to the registry.
+            // Since we can't do this via the RegistryKey APIs,
+            // do it manually via P/Invoke calls.
+
+            const string TestValueName = "CorruptData";
+            string corrupt = "str1\0str2\0str3\0";
+
+            var handle = TestRegistryKey.Handle;
+
+            int ret = Interop.mincore.RegSetValueEx(handle, TestValueName, 0,
+                RegistryValueKind.MultiString, corrupt, corrupt.Length * 2); // Don't include final null-terminator of .NET strings
+
+            if (ret != 0)
+                Assert.True(false, Helpers.GetWin32ErrorMessage(ret));
```

Comment:
You could also just do:
```C#
Assert.Equal(0, ret);
```
and not worry about translating to an error message; if it ever fails (which should be rare to never), someone can look up what the error code means.